src/app/user/shared/authentication.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from "@angular/core";
+import { Router } from "@angular/router";
+import { BehaviorSubject, Observable } from "rxjs";
+import { IUser } from "./user.model";
+
+@Injectable()
+export class AuthService {
+  currentUser:IUser = {
+    username: '',
+    password: '',
+    firstName: '',
+    lastName: '',
+    address: [],
+    pictureURL: ''
+  }
+  private isLoggedIn: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+  isLoggedIn$: Observable<boolean> = this.isLoggedIn.asObservable();
+
+  
+  constructor(private router:Router) {}
+
+
+  loginUser(username:string, password:string) {
```

Comment:
Typically, you'll have some type of return type here. Maybe you get back user information or claims and this is usually an observable. No change here though.